File: api/Entities/Finance/VoucherEntry.cs
using System.ComponentModel.DataAnnotations;

namespace api.Entities.Finance
{
    public class VoucherEntry: BaseEntity
    {
     
        public VoucherEntry()
        {
        }

        public int FinanceVoucherId { get; set; }
        public DateTime TransDate { get; set; }
        public int CoaId { get; set; }
        public string AccountName { get; set; }
        public long Dr {get; set;}
        public long Cr {get; set;}
        public string Narration { get; set; }
        public string DrEntryApprovedByUsername { get; set; }
        public DateTime? DrEntryApprovedOn { get; set; }
        public bool? DrEntryApproved { get; set; }
        public string Remarks { get; set; }
    
    }
}

File: client/src/app/_models/finance/voucherEntry.ts


export interface IVoucherEntry{
	
	id: number;
	financeVoucherId: number;
	transDate: Date;
	coaId: number;
	dr: number;
	cr: number;
	accountName: string;
	narration: string;
	drEntryApproved: boolean;
	drEntryApprovedByUsername: string
	drEntryApprovedOn: Date;
}

export class VoucherEntry implements IVoucherEntry
{
	id=0;
	financeVoucherId=0;
	transDate=new Date();
	coaId=0;
	accountName = '';
	dr=0;
	cr=0;
	narration= '';
	requiresApproval=false;
	drEntryApproved = false;
	drEntryApprovedByUsername = '';
	drEntryApprovedOn = new Date();
}